wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+	
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+	
+		// Check if the 'Eck_Collection_Camp1' exists
+		$collectionCampData = $decodedData['Eck_Collection_Camp1'][0]['fields'] ?? null;
+		if ($collectionCampData === null) {
+			return;
+		}
+
+		// Get subtype information
+		$subtypeMap = self::getSubtypeMap();
+	
+		// Access the subtype
+		$subtypeId = $collectionCampData['subtype'] ?? null;
+		$subtypeLabel = $subtypeMap[$subtypeId] ?? null;		
+	
+		// Check if the subtype is 4 or 5 (collection camp and dropping center)
+		if (!in_array($subtypeLabel, ['Collection Camp', 'Dropping Center'])) {
+			return;
+		}
```

Comment:
Do we need this? I don't understand why we want to limit this to only collection camp or dropping center. If in future we introduce more subtype we want to keep generating the code.